src/Microsoft.AspNet.OData.Shared/NavigationPath.cs

```diff
@@ -0,0 +1,90 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="NavigationPath.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Navigation Path of an OData ID
+    /// </summary>
+    public class NavigationPath
+    {
+        private string _navigationPathName;
+        private ReadOnlyCollection<ODataPathSegment> _pathSegments;
+        PathItem[] _pathItems;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NavigationPath"/> class.
+        /// </summary>        
+        /// <param name="pathSegments">Pathsegment collection</param>
+        public NavigationPath(ReadOnlyCollection<ODataPathSegment> pathSegments)
+        {
+             Debug.Assert(pathSegments.Count > 0);
```

Comment:
>Debug.Assert(pathSegments.Count > 0); [](http://example.com/codeflow?start=13&length=37)

if this is a public API, then we should validate (not assert) that pathSegments is non null and has count >0.